src/satellite_images_nso/api/nso_georegion.py

```diff
@@ -166,18 +170,47 @@ def __init__(
 
     def __getFeatures(self, path):
         """
-        Function to parse features from GeoDataFrame in such a manner that rasterio wants them
+        Function to parse features from GeoDataFrame in such a manner that the NSO api wants them.
 
         @param path: The path to a geojson.
         @return coordinates which rasterio wants to have.
         """
+        gdf = gpd.read_file(path)
         json_loaded = json.loads(gpd.read_file(path).to_json())
+        buffered_polygon = False
 
         if json_loaded["features"][0]["geometry"]["type"] == "MultiPolygon":
-            logging.info("Caution multiploygons are not well supported!")
-            print("Caution multiploygons are not well supported!")
+            logging.info("Multipolygon detected!")
+            print("Multipolygon detected!")
+
+            logging.info("Buffering to grow the different multipolygons together")
+            print(
+                "Buffering to grow the different multipolygons together in one polygon for retrieving links"
+            )
+
+            gdf = gdf.set_crs("EPSG:4326").to_crs("EPSG:28992")
+            buffer_x = 0
+            while gdf.geometry.iloc[0].geom_type != "Polygon":
+                gdf["geometry"] = unary_union(gdf["geometry"].buffer(buffer_x))
+                buffer_x = buffer_x + 1
+
+                # Raise a error in case the buffer gets to big
+                if buffer_x > 200:
+                    raise Exception(
+                        "Multipolygon buffer limit exceeded still a no valid geometry, might not be a error but check your geojson."
+                    )
 
-        return [json_loaded["features"][0]["geometry"]["coordinates"]]
+            gdf = gdf.to_crs("EPSG:4326")
+            buffered_polygon = json.loads(gdf.to_json())
+
+        if buffered_polygon is False:
+            return [json_loaded["features"][0]["geometry"]["coordinates"]][
+                0
+            ], buffered_polygon
```

Comment:
I would return the same for the buffered_polygon as for the polygon, instead of suddenly returning a boolean. I suspect that will make other code easier as well.

Reply:
Yes, I'm not saying don't return the original polygon. I'm saying return the original polygon as buffered_polygon instead of a boolean.

Reply:
i don't fully understand we need to have 2  variable if we buffer a multipolygon.

1 which is the buffered polygon for which we will search the NSO api with.  Buffered polygon
2. to crop the resulting satellite image. this have to be the orginal multipolygon

buffered_polygone is morphing between a polygon and a boolean, because we need to check if we are dealing with a multiple polygon. Buffered polygon is false if we are dealing with a single polygon and a array if we are dealing with multipolygons.

it's much easier to use this as boolean in later code to check if we are dealing with a multipolygon